src/models.py

```diff
@@ -88,7 +88,29 @@ def build_model(args, vocab, pretrained_embs, tasks):
     rnn_params = Params({'input_size': d_emb, 'bidirectional': True,
                          'hidden_size': args.d_hid, 'num_layers': args.n_layers_enc})
 
-    if any(isinstance(task, LanguageModelingTask) for task in tasks) or \
+    if args.sent_enc=="onlstm":
+        phrase_layer = ONLSTMStack(
```

Comment:
Is there a reason this is here in particular? It makes sense to follow the LM as much as possible, and the LM doesn't define a phrase layer here.

Reply:
I'm missing something. I don't see anything else in models.py that creates a phrase layer like this.

Reply:
This still seems like an issue.

Reply:
Moved to modules.py, so now it behaves similar to BowSentenceEncoder or the rest. I still need to put the if case before LanguageModelingTask because ON-LSTM will use the same sent_enc irrespective of the task.